BasicSampleKotlin/app/src/main/java/com/android/example/basicsample/kotlin/ProductFragment.kt

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2017, The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.android.example.basicsample.kotlin
+
+import android.arch.lifecycle.Observer
+import android.arch.lifecycle.ViewModelProviders
+import android.databinding.DataBindingUtil
+import android.os.Bundle
+import android.support.v4.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import com.android.example.basicsample.kotlin.databinding.ProductFragmentBinding
+import com.android.example.basicsample.kotlin.model.Comment
+import com.android.example.basicsample.kotlin.ui.CommentAdapter
+import com.android.example.basicsample.kotlin.ui.CommentClickCallback
+import com.android.example.basicsample.kotlin.viewmodel.ProductViewModel
+
+class ProductFragment : Fragment() {
+
+    private var mBinding: ProductFragmentBinding? = null
+
+    private var mCommentAdapter: CommentAdapter? = null
+
+    private val mCommentClickCallback = object : CommentClickCallback {
+        override fun onClick(comment: Comment) {
+            // no-op
+
+        }
+    }
+
+    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?,
+                              savedInstanceState: Bundle?): View? {
+        // Inflate this data binding layout
+        mBinding = DataBindingUtil.inflate(inflater!!, R.layout.product_fragment, container, false)
+
+        // Create and set the adapter for the RecyclerView.
+        mCommentAdapter = CommentAdapter(mCommentClickCallback)
+        mBinding!!.commentList.adapter = mCommentAdapter
```

Comment:
No, mine was cleaner :p

Reply:
PR into a PR into a PR ;P

Reply:
It's all quite out of date, and a rather mechanical conversion of the Java sample. If Zhuinden has put more attention into his I would recommend looking into that one instead.

Reply:
Should we close or update this one?